challenge-22/submissions/paulmarshall/solution-template.go

```
@@ -0,0 +1,88 @@
+	if amount == 0 || len(denominations) == 0 {
+		return 0
+	}
+
+	// Assume denominations ordered
+	coins := 0
+
+	countDenominations := len(denominations)
+	for i := countDenominations - 1; i >= 0; i-- {
+		value := denominations[i]
+		for {
+			if value > amount {
```

golangci-lint (2.12.2)

[error] QF1006: could lift into loop condition

(staticcheck)